### DISCUSSION TITLE
Can't assign ActionRowBuilder to components

### DISCUSSION THREAD

**Discussion by FelicitusNeko** [discordjs/discord.js#8517] [Q&A] [answered]
I'm having an issue with Discord.js v14 where I don't seem to be able to add an action row to a message, despite (AFAIK) doing exactly as the guide says to do.

Error:
```
src/Archipelabot.ts:266:20 - error TS2322: Type 'ActionRowBuilder<AnyComponentBuilder>' is not assignable to type 'APIActionRowComponent<APIMessageActionRowComponent> | JSONEncodable<APIActionRowComponent<APIMessageActionRowComponent>> | ActionRowData<...>'.

266       components: [actRow]
```

Code:
```typescript
const actRow = new ActionRowBuilder().addComponents(
  new ButtonBuilder({
    customId: "accept",
    label: "Accept",
    style: ButtonStyle.Primary,
  }),
  // snip two more buttons
);

const msg = (await receivingUser.send({
  content: `${userMention(sendingUser)} has sent you a YAML. Please review it and choose if you'd like to add it to your collection.`,
  files: [{ attachment: Buffer.from(yamlData.data) }],
  components: [actRow] // here is the error
}));
```

This error happens in every spot in the code where I try to add components to messages or interaction responses.

Using Node v18.7.0 and Discord.js v14.2.0.

**Comment by Jiralite** [comment, marked as answer]
I assume you're reading from https://discordjs.guide/interactions/buttons.html since I see buttons. There is one thing you are not doing which is dictated for TypeScript users:

<img width="1153" alt="image" src="https://user-images.githubusercontent.com/33201955/185343509-c201b655-00c2-40cd-bc06-97b251381a66.png">

**Reply by FelicitusNeko** [reply]
Yup, that did it. Thanks. Sometimes, it just takes a second set of eyes to catch the problem.
